docs/src/main/asciidoc/infinispan-client-reference.adoc

```diff
@@ -506,9 +542,28 @@ the field, constructor or method. In the below code we utilize field and constru
     RemoteCacheManager remoteCacheManager;
 ----
 
-If you notice the `RemoteCache` declaration has an additional optional annotation named `Remote`.
-This is a qualifier annotation allowing you to specify which named cache that will be injected. This
+If you notice the `RemoteCache` declaration has an additional annotation named `Remote`.
+This is a *qualifier* annotation allowing you to specify which named cache that will be injected. This
 annotation is not required and if it is not supplied, the default cache will be injected.
+The RemoteCacheManager bean scope is `@ApplicationScope`.
+The RemoteCache bean scope is `@Singleton`.
+
+For non default connections, combine the qualifier `@InfinispanClientName` and `@Remote`.
+
+.SomeClass.java
+[source,java]
+----
+    @Inject
+    @InfinispanClientName("lon-site")
+    @Remote("books")
```

Comment:
As this is most definitely hard to use for users and very likely to introduce bugs due to misconfiguration, maybe we should propose using meta-annotations (I assume they work, but @manovotn can tell us for sure) in the documentation?

Reply:
Not sure I see what you mean - if you want CDI to read and work with the annotation, then it'd need to be a [CDI stereotype](https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0.html#stereotypes). However, those cannot contain bean qualifiers which makes it unusable in this case (AFAICS, `@Remote` is a qualifer, not sure about `@InfinispanClientName`)

Though you could have a processor that reads the newly added meta annotation and then adds actual CDI annotations programmatically. But that has the downside of adding a behind the scenes magic.

Reply:
@geoand I don't want to touch the Remote annotation and add the client name on it. I rather leave two qualifiers, since having named connections won't be the most common use case.
However, I do agree that injecting RemoteCache directly with a qualifier might be confusing and programatically injecting the structure entry points (RemoteCacheManager, CounterManager ... or the new api entry point).

Reply:
Understood